### PR TITLE
Use crevice to write LightsUniform

### PR DESCRIPTION
**Pull request by mtsr**
This is a draft of what usage of crevice for writing Std140 layout looks like, see #1779.

Note that `WriteStd140` is fairly low overhead, it converts glam types into `Std140` primitives before serializing). `AsStd140`, which can be derived in many cases, potentially has more overhead, since it converts the entire struct into a version with padding and `Std140` primitives.

Dynamic sizing of `WriteStd140` is also not without cost, since it serializes the type to `io::Sink()` and uses `writer.len()` to calculate the size.

This draft currently depends on an upcoming PR on crevice that adds the `StaticStd140Size` trait, so that the maximum size of `LightsUniform` can be calculated.

Additionally there's draft PR to glam to implement crevice traits. This is probably the most problematic part of crevice right now. Because of the orphan rule it's impossible to implement crevice traits without newtype wrappers on foreign types. There might be an upcoming solution, Lucien has been very welcoming in discussing improvements.

### CODE REVIEW

File: crates/bevy_math/Cargo.toml [hunk 1]
```diff
@@ -13,5 +13,12 @@ license = "MIT"
 keywords = ["bevy"]
 
 [dependencies]
-glam = { version = "0.13.0", features = ["serde"] }
-bevy_reflect = { path = "../bevy_reflect", version = "0.5.0", features = ["bevy"] }
+glam = { version = "0.14.0", features = ["serde"], path = "../../../glam-rs" }
+crevice = { version = "0.6", optional = true, path = "../../../crevice" }
```
Comment (mockersf) [inline crates/bevy_math/Cargo.toml:17]: could you put your forks of those on GitHub and point to them?

**Comment by mockersf** [comment]
I'm very interested in this to keep backends other than the official one (mainly WebGL2) working.

For example, the WebGL2 shader currently fails after #1778 got merged, even after applying the changes, to do a wrongly sized buffer

**Comment by mtsr** [comment]
> 
> 
> I'm very interested in this to keep backends other than the official one (mainly WebGL2) working.
> 
> For example, the WebGL2 shader currently fails after #1778 got merged, even after applying the changes, due to a wrongly sized buffer

Crap, that sucks. I need to make testing webgl part of my workflow for any future changes to shaders.

File: crates/bevy_math/Cargo.toml (hunk 1 above)
Comment (mtsr) [reply crates/bevy_math/Cargo.toml]: Done.

**Comment by james7132** [comment]
Closing this now that `bevy_crevice` is used for all of our rendering pipelines.

**Pull request closed without merging**